druid/src/core.rs

```diff
@@ -266,18 +266,14 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
             base_state: &mut self.state,
             had_active,
             is_handled: false,
-            is_root: false,
         };
         let rect = child_ctx.base_state.layout_rect;
         // Note: could also represent this as `Option<Event>`.
         let mut recurse = true;
         let mut hot_changed = None;
         let child_event = match event {
             Event::LifeCycle(event) => Event::LifeCycle(*event),
-            Event::Size(size) => {
-                recurse = ctx.is_root;
-                Event::Size(*size)
-            }
+            Event::Size(size) => Event::Size(*size),
```

Comment:
Size events propagate to children. In the case of the VirtualList, the max_scroll_position is calculated prior to the layout run so the scrollbar can layout the thumb in relation to the virtualized viewport size. 

This could alternately be accomplished if the `Data` were mutable in `layout` as well but I understand the purpose of the layout run is to commit positioning for `paint`. Mutating data in layout could cause side effects.